core/src/main/java/com/sequenceiq/cloudbreak/service/stack/CmCommandLinkProvider.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.cloudbreak.service.stack;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.cluster.gateway.topology.ClusterExposedServiceV4Response;
+import com.sequenceiq.cloudbreak.api.service.ExposedServiceCollector;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.service.ServiceEndpointCollector;
+import com.sequenceiq.cloudbreak.util.StackUtil;
+
+@Service
+public class CmCommandLinkProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CmCommandLinkProvider.class);
+
+    @Value("${distrox.gateway.topology.name}")
+    private String defaultTopologyName;
```

Comment:
Is this really relevant for DH only? If not (i.e. it also applies for DL stacks), I wonder if the `distrox` category is indeed appropriate.

Reply:
no, it is relevant for datalake too and it is applied for datalake too, the config name is misleading, but I do not want to change it in this PR